model_templates/training/python3_pytorch_multiclass/custom.py

```diff
@@ -0,0 +1,115 @@
+import os
+import pickle
+from typing import List, Optional, Any
+
+import numpy as np
+import pandas as pd
+from sklearn.preprocessing import LabelEncoder
+import torch
+
+from preprocessing import dense_preprocessing_pipeline
+from model_utils import build_classifier, train_classifier, save_torch_model
+
+preprocessor = None
+
+
+def load_model(code_dir: str) -> Any:
+    """
+    Can be used to load supported models if your model has multiple artifacts, or for loading
+    models that **drum** does not natively support
+
+    Parameters
+    ----------
+    code_dir : is the directory where model artifact and additional code are provided, passed in
+
+    Returns
+    -------
+    If used, this hook must return a non-None value
+    """
+    global preprocessor
+    with open(os.path.join(code_dir, "preprocessor.pkl"), mode="rb") as f:
+        preprocessor = pickle.load(f)
+
+    model = torch.load(os.path.join(code_dir, "artifact.pth"))
+    model.eval()
+    return model
+
+
+def transform(data: pd.DataFrame, model: Any) -> pd.DataFrame:
+    """
+    Intended to apply transformations to the prediction data before making predictions. This is
+    most useful if **drum** supports the model's library, but your model requires additional data
+    processing before it can make predictions
+
+    Parameters
+    ----------
+    data : is the dataframe given to **drum** to make predictions on
+    model : is the deserialized model loaded by **drum** or by `load_model`, if supplied
+
+    Returns
+    -------
+    Transformed data
+    """
+    if preprocessor is None:
```

Comment:
oof that is notgreat

Reply:
?

Reply:
cant you pass it through the return for load_model.

Reply:
That'd require me to then implement my own hook for `score`

Reply:
which to me feels like overkill in this case

Reply:
alright. I still dont like that we're advertising this as best practice though right.

Reply:
i get that. This seemed like the least awful approach. Other option I thought of was to load the model every time transform is called. Pros: no globals. Cons: expensive and slow

Reply:
I feel like this might have been a design mistake on our part to not have users build python classes instead of calling python methods.

Reply:
that may be, but is outside the scope of this PR. perhaps standalone hooks can deprecated in favor of a new class based approach in a major version of drum